src/redis/cache_impl.go

```diff
@@ -134,7 +134,7 @@ func (this *rateLimitCacheImpl) DoLimit(
 	}
 
 	// request.HitsAddend could be 0 (default value) if not specified by the caller in the Ratelimit request.
-	hitsAddend := max(1, request.HitsAddend)
+	hitsAddend := max(0, request.HitsAddend)
```

Comment:
If request.HitsAddend is zero, should we just return instead of continuing the process?

Reply:
Agree, this change would break existing callers who expect the default of 1 behavior.

Reply:
We want the call to go through to see what the current value is in the redis cache for the key.

This change would make setting hitsAddend as required key in proto (need to set `required = true` using [protoc-gen-validate](https://github.com/envoyproxy/protoc-gen-validate)). Current behavior is that we'd add 1 if hitsAddend is missed in ratelimit request.

Reply:
That would break backwards compatibility of the proto, which should be treated as the last resort for adding or changing existing functionality.